### ISSUE TITLE
Can Softmax Loss Predict learn multiple classes?

### ISSUE THREAD

**Issue by ylqfp** [fchollet/keras#2564]
Such as, target Y have multiple value, like [0,1,0,1,0,1,0]?
Thanks!

**Comment by braingineer** [comment]
softmax normalizes so that the outputs sum to 1.    You can switch to a sigmoid activation and do multiclass classification.

**Comment by ylqfp** [comment]
Thanks, I've not described clearly, Graph container is what I want.
Thanks all the same!

**Comment by braingineer** [comment]
so you meant multiple binary outputs?

**Comment by ylqfp** [comment]
Let me clarify.
I'm considering multitask sceenario, which needs multiple inputs and multiple outputs.
Also inside the process, there will be branches split and merge.
Generally speaking, It's rather a shallow but complecated model.
The output I want is : distribution on label set 1 and distribution on label set 2, and their score and their weight, and different loss on the two different sets. 
When I got the score, I could transform them to binary if i want.

**Comment by braingineer** [comment]
Ah. that's nothing like what you originally said =P. 

Don't use Graph, though.  Use the functional API.  you can just as easily have multiple outputs and inputs. 

```python
input1 = Input(shape=whatever)
input2 = Input(shape=whatever)

func1 = Dense(hid_num) ### this only creates the dense function

out1 = func1(input1)
out2 = func1(input2)

func2 = Dense(hid_num)

out3 = func2(out1)

## you can keep doing things like this until
model = Model(input=[input1, input2], output=[out1, out2, out3])
```
you can see more here: http://keras.io/getting-started/functional-api-guide/

**Comment by ylqfp** [comment]
@braingineer So much thankful for your valuable advice!
However, could I ask why don't you suggest use Graph? Not so mature, or difficult to use?
Cause I only know it's flexible and powerful to model structure.

**Comment by braingineer** [comment]
It's not modern, it's deprecated. that's why it's under `legacy` : https://github.com/fchollet/keras/blob/master/keras/legacy/models.py#L10

The functional API uses an underlying representation of a graph now.  The code above is a graph with multiple inputs and outputs.  http://keras.io/getting-started/functional-api-guide/#the-concept-of-layer-node

**Comment by ylqfp** [comment]
I'll study the material carefully, thanks!!
